### PR TITLE
[Chat] Adding new apis to ChatAdapter

### PR DESCRIPTION
**Pull request by JoshuaLai**
# What
Exposing the new apis on the chat stateful client

# Why
Exposing the new apis on the chat stateful client

# How Tested

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/communication-react/review/beta/communication-react.api.md [hunk 1]
```diff
@@ -1007,6 +1007,8 @@ export interface CallWithChatAdapterManagement {
     downloadAttachment: (options: {
         attachmentUrl: string;
     }) => Promise<AttachmentDownloadResult>;
+    // (undocumented)
```
Comment (Leah-Xia-Microsoft) [inline packages/communication-react/review/beta/communication-react.api.md:1010]: Should we add some documentation here?

File: packages/react-composites/src/mocks/MockChatClient.ts [hunk 2]
```diff
@@ -2,21 +2,24 @@
...
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { ChatClientState, ChatErrors } from '@internal/chat-stateful-client';
 import { PagedAsyncIterableIterator } from '@azure/core-paging';
 
 type PublicInterface<T> = { [K in keyof T]: T[K] };
 /**
  * @returns
  */
-export const createStatefulChatClientMock = (threadClient: PublicInterface<ChatThreadClient>): StatefulChatClient => {
+export function createStatefulChatClientMock(threadClient: PublicInterface<ChatThreadClient>): StatefulChatClient {
```
Comment (Leah-Xia-Microsoft) [inline packages/react-composites/src/mocks/MockChatClient.ts:20]: Any reason for not using the arrow function syntax here?

**Review by Leah-Xia-Microsoft — APPROVED** [review]

File: packages/react-composites/src/mocks/MockChatClient.ts (hunk 2 above)
Comment (JoshuaLai) [reply packages/react-composites/src/mocks/MockChatClient.ts]: Reverted

File: packages/communication-react/review/beta/communication-react.api.md (hunk 1 above)
Comment (JoshuaLai) [reply packages/communication-react/review/beta/communication-react.api.md:1010]: This just needed to be regenerated but the documentation is already in place

File: packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts [hunk 3]
```diff
@@ -480,9 +484,18 @@ export class AzureCommunicationCallWithChatAdapter implements CallWithChatAdapte
     this.chatAdapter.updateFileUploadMetadata(id, metadata);
   };
   /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+  // ToDo: This method is to be removed
   async downloadAttachment(options: { attachmentUrl: string }): Promise<AttachmentDownloadResult> {
     return await this.chatAdapter.downloadAttachment(options);
   }
+  /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+  public downloadResourceToCache(threadId: string, messageId: string, resourceUrl: string): void {
```
Comment (emlynmac) [inline packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts:492]: Shall we put these three parameters in one object for clarity and ease of use?

**Review by emlynmac — COMMENTED** [review]
Couple of thoughts for the method parameters.
Let's get the linter / CC issues fixed up too.

File: packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts (hunk 3 above)
Comment (JoshuaLai) [reply packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts:494]: I am thinking actually 2 arguments. Something like `(threadId: string, {messageId: string, resourceUrl: string})` where the second argument encapsulates message specific properties.

File: packages/react-composites/review/beta/react-composites.api.md [hunk 4]
```diff
@@ -1140,9 +1152,19 @@ export interface ChatAdapterThreadManagement {
...
     }) => Promise<AttachmentDownloadResult>;
+    downloadResourceToCache(resourceDetails: {
+        threadId: string;
+        messageId: string;
+        resourceUrl: string;
+    }): void;
     fetchInitialData(): Promise<void>;
     loadPreviousChatMessages(messagesToLoad: number): Promise<boolean>;
     removeParticipant(userId: string): Promise<void>;
+    removeResourceFromCache(resourceDetails: {
```
Comment (emlynmac) [inline packages/react-composites/review/beta/react-composites.api.md:1163]: Would be good to export this as a type for reuse

**Review by emlynmac — APPROVED** [review]

File: packages/react-composites/review/beta/react-composites.api.md (hunk 4 above)
Comment (JoshuaLai) [reply packages/react-composites/review/beta/react-composites.api.md]: Done

**Pull request merged**